src/main.c

```diff
@@ -188,48 +192,50 @@ version(void)
 void
 get_choices(void)
 {
-	char		*line, *description, *field_separators;
-	size_t		 line_size;
+	char		*description, *field_separators, *start, *stop;
 	ssize_t		 length;
 	struct choice	*choice;
 
 	if ((field_separators = getenv("IFS")) == NULL)
 		field_separators = " ";
 
+	input.size = BUFSIZ;
+	if ((input.v = malloc(input.size)) == NULL)
+		err(1, "malloc");
+	for (;;) {
+		if ((length = read(0, input.v + input.nmemb,
```

Comment:
Fixed. Not a fan but it feels more consistent with the current codebase.

On Tue, Dec 22, 2015 at 01:27:09AM -0800, Calle Erlandsson wrote:
> >  	ssize_t		 length;
> >  	struct choice	*choice;
> >  
> >  	if ((field_separators = getenv("IFS")) == NULL)
> >  		field_separators = " ";
> >  
> > +	input.size = BUFSIZ;
> > +	if ((input.v = malloc(input.size)) == NULL)
> > +		err(1, "malloc");
> > +	for (;;) {
> > +		if ((length = read(0, input.v + input.nmemb,
> 
> Do you think replacing `0` with `STDIN_FILENO` would be make this easier to read?
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/thoughtbot/pick/pull/81/files#r48235527

-- 
:wq